test/ERC20LockUpScenarios.test.ts

```diff
@@ -0,0 +1,328 @@
+      const alinasFinalRewards = await poolContract.pendingRewards(alina.address);
+      let alinaFirstClaimBalance = await mockRewardToken.balanceOf(alina.address);
+      expect(alinasFinalRewards).to.be.gt(0); // Alina should have final rewards
+      await poolContract.connect(alina).claim();
+      let alinaBalance = await mockRewardToken.balanceOf(alina.address);
+      expect(alinaBalance).to.be.equal(alinaFirstClaimBalance + alinasFinalRewards);
+    });
+  
+  
+    it("Handles multiple users staking different amounts, unstaking, and claiming", async function () {
+      poolStartTime = (await time.latest()) + 100; // Start in 100 seconds
+      poolEndTime = poolStartTime + 1000; // End in 1000 seconds after start
+      unstakeLockUp = poolStartTime + 500; // Lockup 500 seconds after start
+      claimLockUp = poolStartTime + 200; // Lockup 200 seconds after start
+      rewardTokenPerSecond = ethers.parseEther("1"); // 1 RWD per second
+      const data = {
+        stakeToken: await mockStakeToken.getAddress(),
+        rewardToken: await mockRewardToken.getAddress(),
+        poolStartTime: poolStartTime,
+        poolEndTime: poolEndTime,
+        unstakeLockUpTime: unstakeLockUp,
+        claimLockUpTime: claimLockUp,
+        rewardPerSecond: rewardTokenPerSecond
+      };
+      const { poolContract } = await deployAndSetupPool(ipfsHash, ayo, data);
+  
+  
+      const users = [ayo, alina, vartan, signer, nikita, mary, mansur];
+      let poolAddress = await poolContract.getAddress();
+      for (const user of users) {
+        await mockStakeToken.mint(user, ethers.parseEther("10000000000"));
+        await approve(1000000000, user, mockStakeToken, poolAddress)
+      }
+  
+      await time.increaseTo(poolStartTime);
+  
+      await poolContract.connect(ayo).stake(ethers.parseEther("75"));
+      await poolContract.connect(alina).stake(ethers.parseEther("120"));
+      await poolContract.connect(vartan).stake(ethers.parseEther("90"));
+      await poolContract.connect(signer).stake(ethers.parseEther("115"));
+      await poolContract.connect(nikita).stake(ethers.parseEther("85"));
+      await poolContract.connect(mary).stake(ethers.parseEther("105"));
+  
+      await time.increaseTo(unstakeLockUp);
+  
+      await poolContract.connect(ayo).unstake(ethers.parseEther("30"));
+      await poolContract.connect(alina).unstake(ethers.parseEther("40"));
+      await poolContract.connect(vartan).unstake(ethers.parseEther("50"));
+      await poolContract.connect(signer).unstake(ethers.parseEther("60"));
+      await poolContract.connect(nikita).unstake(ethers.parseEther("25"));
+      await poolContract.connect(mary).unstake(ethers.parseEther("35"));
+  
+      const ayoInitialBalance = await mockRewardToken.balanceOf(ayo.address);
+      await poolContract.connect(ayo).claim();
+      const ayoFinalBalance = await mockRewardToken.balanceOf(ayo.address);
+      expect(ayoFinalBalance).to.be.gt(ayoInitialBalance);
```

Comment:
as I described in the comment above, "gt" is not appropriate, only "equal" should be used everywhere. Sometimes "closeTo" can also be used with very very small approximation, if there is no way to calculate an exact result value